src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -195,6 +195,18 @@ private function parseDefinition($id, $service, $file)
             $definition->setAbstract($service['abstract']);
         }
 
+        if (isset($service['deprecated'])) {
+            if (is_bool($service['deprecated'])) {
+                $definition->setDeprecated($service['deprecated']);
+            } elseif (is_array($service['deprecated']) && isset($service['deprecated']['status'])) {
+                if (isset($service['deprecated']['message'])) {
+                    $definition->setDeprecated($service['deprecated']['status'], $service['deprecated']['message']);
+                } else {
+                    $definition->setDeprecated($service['deprecated']['status']);
+                }
+            }
```

Comment:
this API is not consistent with the way other features are configured in YAML (for instance service decoration). The following would be more consistent:

```yaml
services:
    foobar:
        # ...
        deprecated: true
        deprecation_message: "..."
```